dash_agent/README.md

```diff
@@ -1,39 +1,174 @@
+
+  @override
+  List<Command> get registerSupportedCommands =>
+      [AskCommand(docsSource: docsSource)];
+}
+```
+
+The above sample AgentConfiguration object taking data sources and supported commands. Both of them are explained below:
+
+- `DataSource`: Data sources will enable you to pass on any form of data that your agent might need to perform its intended tasks.
+- `Command`: Commands are the specialised tasks you want your agents to perform (like understanding the developer query, refactoring, code generation, etc).
+
+### Datasource
+
+As described above data sources lets you attach the data that your agent will need to perform its tasks. It can be anything from raw texts, json, files data, or even webpages.
+
+Sample example for DataSource for demostration purpose:
+
+```dart
+class DocsDataSource extends DataSource {
+
+ /// Enables you to pass data stored in files and directories in you local system.
+ @override
+ List<FileDataObject> get fileObjects => [
+       FileDataObject.fromFile(File(
+           'your_file_path')),
+        FileDataObject.fromDirectory(Directory(
+            'directory_path_to_data_source'))
+     ];
+ 
+
+ /// Enables you to pass in raw string and json data
+ @override
+ List<ProjectDataObject> get projectObjects =>
+     [ProjectDataObject.fromText('Data in form of raw text')];
+
+
+ /// Enables you to pass web data your agent by passing web pages urls or sitemaps in the object.
+  @override
+ List<WebDataObject> get webObjects =>
+     [WebDataObject.fromWebPage('https://sampleurl.com'), 
+     WebDataObject.fromSiteMap('https://sampleurl.com/sitemap.xml')];
+}
+```
+
+To create any data object, the easiest (and also the recommended way) is to call static functions of the above shared bases class of the object. For example if you want to store the json data. You can add the json data as shared below:
 
 ```dart
-const like = 'sample';
+final yourJson = {'key': 'data'};
+final jsonDataSource = ProjectDataSource.fromJson(yourJson);
 ```
 
+**Note**: At the moment, storing pdf files is not supported out of the box. However you can extract the relevant data from the pdf using the open source tools and pass the extracted data either via project data object or system data object.
+
+### Commands
+
+Commands are the specialised tasks you want your agents to perform  (such as refactoring, code generation, code analysis, etc). Once the agent will be published, user's can invoke this command in the commanddash client (such as VS Code extensions) and use them. 
+
+Sample example for `Command` object is shared below:
+
+```dart
+class AskCommand extends Command {
+  AskCommand({required this.docsSource});
+
+  final DataSource docsSource;
+
+  // Inputs
+  final userQuery = StringInput('Your query');
+  final codeAttachment = CodeInput('Code Attachment');
+
+ // Outputs
+ final matchingDocuments = MatchDocumentObject();
+ final queryOutput = QueryOutput();
```

Comment:
The outputs should be defined and used within the scopes of `steps` only. 

This is our recommended practice since need of outputs is limited to the execution of steps.

Reply:
Sure thing. I will make the update